### PR TITLE
EC-76 generating mobile specific deep link

### PR DESCRIPTION
**Pull request by darshan-patel**
@edcastrohit @sagarbhute @yogendra689 @amardaxini @akshayEdCast @zhifeng-huang @NileshN 

Generating a mobile-specific deep link and pushing it to ECL.

To identify file mime-type, I used Apache-tika jar(https://tika.apache.org/1.17/gettingstarted.html).
Apache-tika jar needs java environment.

Kindly review.

### CODE REVIEW

File: app/service/content_extraction_service.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module ContentExtractionService
+  
+  def get_file_mime_type(file_path)
+    begin
+      `#{java} -jar #{Rails.root.to_path}/vendor/tika-app-1.14.jar -d "#{file_path}"`
```
Comment (amardaxini) [inline app/service/content_extraction_service.rb]: Can you share output here? Is there any ruby library is available?

Comment (amardaxini) [reply app/service/content_extraction_service.rb]: If file is 1GB  what would be the performance impact.

Comment (amardaxini) [reply app/service/content_extraction_service.rb]: Can you share file path for reference,Can you add test cases for this

Comment (amardaxini) [reply app/service/content_extraction_service.rb]: If we know the extension we can save one extra process.
mime_type = Rack::Mime.mime_type(extension)

Comment (darshan-patel) [reply app/service/content_extraction_service.rb]: SharePoint is giving file extension, so I removed this code in commit https://github.com/Course-Master/collector_app/pull/16/commits/e753468bf005bcd76f26a7993bffb1e4e24fcea9

File: app/service/sharepoint_integration.rb [hunk 2]
```diff
@@ -93,6 +110,7 @@ def create_content_item(entry)
         url:          entry["webUrl"]
       },
       additional_metadata: {
+        mobile_link:     generate_mobile_link(entry),
```
Comment (yogendra689) [inline app/service/sharepoint_integration.rb]: No need to add `mobile_link`. Just override url with `generate_mobile_link(entry)`.

Comment (darshan-patel) [reply app/service/sharepoint_integration.rb]: Removed `mobile_link` in commit https://github.com/Course-Master/collector_app/pull/16/commits/02fed3530adbddbcb70a5931e188e8aded98b388

**Pull request merged**